File: subscriptions/urls.py
from django.urls import path, include
from rest_framework.routers import DefaultRouter

from .api import TitleViewSet, SubscriptionsViewSet, SearchView, SubscribeView, HomeView, UnsubscribeView

router = DefaultRouter()
router.register('title', TitleViewSet)
router.register('subscriptions', SubscriptionsViewSet)

urlpatterns = [
    path('', include(router.urls)),
    path('subscribe/<int:content_id>', SubscribeView.as_view()),
    path('unsubscribe/<int:content_id>', UnsubscribeView.as_view()),
    path('search', SearchView.as_view()),
    path('home', HomeView.as_view())
]

File: frontend/src/components/common/urls.js
const RouterUrls = Object.freeze({
    HOME: ['/home', '/'],
    SUBSCRIPTIONS: '/list',
    TITLE: '/title/:id',
    LOGIN: '/login',
    REGISTER: '/register',
    SEARCH: '/search',
});

const getTitleUrl = (id) => {
    return RouterUrls.TITLE.replace(':id', id);
};

const getHomeUrl = () => {
    return RouterUrls.HOME[0];
};

const getSearchUrl = (q) => {
    return RouterUrls.SEARCH + `?q=${q}`;
};

export { RouterUrls, getTitleUrl, getHomeUrl, getSearchUrl };
